doc/doc-3301.md

```diff
@@ -0,0 +1,20 @@
+# `map` template function
+
+# Short description
+
+`map` template function applies a template function to all elements of a list. It returns the new list.
+
+# Config snippet
+
+```
+log {
+  source { example-msg-generator(template("0,1,2") num(1)); };
+  destination { file("/dev/stdout" template("$(map $(+ 1 $_) $MSG)\n")); };
+};
+```
+
+# Description
+
+`map` template function has two mandatory parameters. The first parameter is a template, that will be applied to each list elements. The second parameter is a string or template, that represents the list.
+
+You can use `$_` template to refer to the list element inside the template function.
```

Comment:
[optional]
As @MrAnno suggested nested `map` also works, and would worth to mention. Also `$(map $(map ... ) ..)` works, without lambda probably not useful, but a statement regarding how the `$_` binds (which map it refers to) would be nice to have.

Reply:
Fixed thanks.